docs/alternator/alternator.md

```diff
@@ -129,6 +129,28 @@ isolation policy for a specific table can be overridden by tagging the table
    Example: in order to query the contents of Scylla's system.large_rows,
    pass TableName='.scylla.alternator.system.large_rows' to a Query/Scan request.
 
+### Tag-based default time to live (TTL)
+ * Aside from the DynamoDB TTL mechanism based on the UpdateTimeToLive method,
+   alternator also exposes a way of defining default per-table time to live
+   parameter, expressed in seconds and accessible via a `system:default_time_to_live`
+   tag. The default time to live mechanism will automatically apply TTL
+   on all *new* updated and inserts, and remove them once their expiration period
+   has passed. Note a few important details:
+     - expiration induced by default TTL **does not** produce any Streams events
+     - TTL will only be applied to new updates and inserts, it won't amend existing data in any way
```

Comment:
Perhaps we should be more explicit here on how this works. I can also send a documentation patch later if you want, but @eliransin is worried that if we describe exactly how this feature works - @tzach and users might complain that this is a bad feature and bad implementation:

The thing is, as this is currently implemented, if UpdateItem (as opposed to PutItem) is used to modify just a single (top-level) attribute, only this attribute will get a a new lease on life, while the rest of the item will expire at its old pre-destined time. The result is that at some point in the future, we'll end up with a partial item, with only the modified attribute and nothing else :-) Presumably this state of affairs makes sense to Scylla users (because Scylla has this feature, and people apparently use it?) but no DynamoDB user will ever guess that this is the case.

We have several alternatives on how to "solve" this issue:
1. Document it (with a bit more detail than you did above) and tell users - "caveat emptor!"
2. Forbid UpdateItem when default-ttl is set
3. Cause UpdateItem to always use read-before-write when default-ttl is set, and always rewrite the entire item.
4. Some sort of combination between 2 and 3 depending on configuration options (a new one or the old write isolation option).

Alternative 1 is easiest, and most "Scylla-like" (Scylla also allows this weird partial-item expiration) but I don't know if @eliransin and @tzach will accept it.

Reply:
Another difference from DynamoDB's TTL feature that should probably be mentioned somewhere here is that (like Scylla's TTL), the expiration times here have one second accuracy (the item is deleted within one second of expiring).